### PR TITLE
Add engagement links so we can figure out how users interacted with e…

### PR DESCRIPTION
**Pull request by kaihendry**
…mail notifications

Refactored resolveServiceDomain

Fixes https://github.com/unee-t/frontend/issues/345

### CODE REVIEW

File: imports/email-templates/components/helpers.js [hunk 1]
```diff
@@ -15,13 +17,9 @@ function engage (params) {
   return URL.format(engagementURL)
 }
 
-// Does not work on localhost domains, since we assume 4 parts: $service.$stage.unee-t.com
 function resolveServiceDomain (service) {
```
Comment (nbiton) [inline imports/email-templates/components/helpers.js]: Shouldn't this be in a separate utility module that's not specific to email?

File: imports/email-templates/components/helpers.js [hunk 2]
```diff
@@ -15,13 +17,9 @@ function engage (params) {
...
-  const url = URL.parse(process.env.ROOT_URL)
-  const hparts = url.hostname.split('.')
-  hparts[0] = service
-  url.host = hparts.join('.')
-  return URL.format(url)
+  const domain = process.env.STAGE ? `${service}.${process.env.STAGE}.${process.env.DOMAIN}` : `${service}.${process.env.DOMAIN}`
```
Comment (nbiton) [inline imports/email-templates/components/helpers.js]: We haven't introduced STAGE and DOMAIN to master yet afaik

File: imports/email-templates/components/helpers.js [hunk 3]
```diff
@@ -6,7 +6,9 @@ export function resolveUserName (user) {
   return `${user.profile.name || user.emails[0].address.split('@')[0]}`
 }
 
-function engage (params) {
+// We can use this to actually see how users engaged with emails
+// https://github.com/unee-t/engagement
+export function engage (params) {
```
Comment (nbiton) [inline imports/email-templates/components/helpers.js]: Can you rename it to "createEngagementLink" or something like that?

File: imports/email-templates/components/helpers.js (hunk 2 above)
Comment (kaihendry) [reply imports/email-templates/components/helpers.js]: Actually, I just checked, they are there.

**Pull request merged**